heron/api/src/java/BUILD

```diff
@@ -18,16 +18,16 @@ java_library(
 )
 
 java_binary(
-    name = "api-unshaded",
+    name = "api",
     srcs = glob(["com/twitter/heron/api/**/*.java"]),
     deps = api_deps_files,
 )
 
-genrule(
+jarjar_binary(
     name = "heron-api",
-    srcs = [":api-unshaded_deploy.jar"],
-    outs = ["heron-api.jar"],
-    cmd  = "cp $< $@",
+    src = ":api_deploy.jar",
+    shade = "shade.conf",
```

Comment:
I don't think there is need to exclude Guava dependency now (similarly, heron-storm.jar has Guava dependency that is impossible to remove. so I have to bundle and shade it). That's why I need to shade all `com.google.*`.